src/v/cluster/errc.h

```diff
@@ -59,6 +59,7 @@ enum class errc : int16_t {
     error_collecting_health_report,
     leadership_changed,
     feature_disabled,
+    out_of_disk_space,
```

Comment:
it doesn't seem like this error code is used. i'm not too sure it will be needed, either, right?

Reply:
Yep this will be removed if not needed before I mark this PR ready.. I'm just waiting to finish the new write rejection codepath to make sure I won't need it.

Reply:
This patch has been removed.